web/src/features/comments/CommentList.tsx

```diff
@@ -0,0 +1,241 @@
+import {
+  Avatar,
+  AvatarFallback,
+  AvatarImage,
+} from "@/src/components/ui/avatar";
+import { Button } from "@/src/components/ui/button";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormMessage,
+} from "@/src/components/ui/form";
+import { Input } from "@/src/components/ui/input";
+import { useHasProjectAccess } from "@/src/features/rbac/utils/checkProjectAccess";
+import { api } from "@/src/utils/api";
+import { cn } from "@/src/utils/tailwind";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { type CommentObjectType, CreateCommentData } from "@langfuse/shared";
+import { ArrowUpToLine, LoaderCircle, Trash } from "lucide-react";
+import { useSession } from "next-auth/react";
+import React, { useEffect, useMemo } from "react";
+import { useForm } from "react-hook-form";
+import { type z } from "zod";
+
+const formatCommentTimestamp = (timestamp: Date): string => {
+  const diffInMs = new Date().getTime() - timestamp.getTime();
+  const diffInMinutes = diffInMs / (1000 * 60);
+  const diffInHours = diffInMinutes / 60;
+  const diffInDays = diffInHours / 24;
+
+  if (diffInHours < 1) {
+    return `${Math.floor(diffInMinutes)} minutes ago`;
+  } else if (diffInHours < 24) {
+    return `${Math.floor(diffInHours)} hours ago`;
+  } else if (diffInDays < 7) {
+    return `${Math.floor(diffInDays)} days ago`;
+  } else {
+    return timestamp.toLocaleDateString("en-US", {
+      year: "2-digit",
+      month: "numeric",
+      day: "numeric",
+    });
+  }
+};
+
+export function CommentList({
+  projectId,
+  objectId,
+  objectType,
+  isVisible,
+  className,
+}: {
+  projectId: string;
+  objectId: string;
+  objectType: CommentObjectType;
+  isVisible: boolean;
+  className?: string;
+}) {
+  const session = useSession();
+  const hasReadAccess = useHasProjectAccess({
+    projectId,
+    scope: "comments:read",
+  });
+
+  const hasWriteAccess = useHasProjectAccess({
+    projectId,
+    scope: "comments:CUD",
+  });
+
+  const comments = api.comments.getByObjectId.useQuery(
+    {
+      projectId,
+      objectId,
+      objectType,
+    },
+    { enabled: hasReadAccess && isVisible },
+  );
+
+  const form = useForm<z.infer<typeof CreateCommentData>>({
+    resolver: zodResolver(CreateCommentData),
+    defaultValues: {
+      content: "",
+      projectId,
+      objectId,
+      objectType,
+    },
+  });
+
+  useEffect(() => {
+    form.reset({ content: "", projectId, objectId, objectType });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [objectId, objectType]);
+
+  const utils = api.useUtils();
+
+  const createCommentMutation = api.comments.create.useMutation({
+    onError: (error) => form.setError("content", { message: error.message }),
```

Comment:
does not work as you reset the form onSettled. Suggestion:
- move the invalidation/reset to onSuccess to make sure the draft message is not deleted if the trpc call does not work
- remove the onError thereby use the global onError trpc handler which will render a notification in the top right